controllers/utils/base_reconciler.go

```diff
@@ -355,12 +360,12 @@ func (r *ReconciliationRunner) GetReconciliationSubject() (res ctrl.Result, rete
 	r.Log.V(1).Info("Getting reconciliation subject.")
 	err := client.IgnoreNotFound(r.K8sClient.Get(r.RequestCtx, r.Request.NamespacedName, r.ReconciliationSubject))
 	if r.ReconciliationSubject.GetName() == "" {
-		r.SetReturnEarly()
```

Comment:
This was non-functional. Again, a very rare case, but I fixed it up here having found the issue in tests.